### ISSUE TITLE
"Network Is Not Available" When trying to connect with Todoist

### ISSUE THREAD

**Issue by jleseane23** [alainm23/planify#645]
**Describe the bug**
At first launch or when trying to connect to Todoist from the preferences, I get "Network Is Not Available Connect to the Internet to connect with Todoist" error message. Title box says "Please enter your credentials." 

**To Reproduce**
1. Launch Planner
2. At the first launch wizard, click on Todoist
3. See error

**Expected behavior**
A box to pop up asking for my Todoist credentials

**Screenshots**
![image](https://user-images.githubusercontent.com/55364887/103162559-17970c00-47c0-11eb-9cd5-10d05b23726f.png)

**Desktop (please complete the following information):**
 - Pop!_OS 20.10
 - Gnome version 3.38.1

**Additional context**
* Tried switching to different networks. Same result
* Tried to make sure a credential window wasn't hidden. Couldn't find anything.
* Downloaded the app from Pop!_Shop
* Tried starting a local database and connecting from the preferences menu. Same result

**Comment by alainm23** [comment]
Can you run Planner from terminal and check the error?
`flatpak run com.github.alainm23.planner`

**Comment by jleseane23** [comment]
I get the following when launching:

_Failed to load module "appmenu-gtk-module"

** (com.github.alainm23.planner:2): CRITICAL **: 16:12:26.537: objects_project_get_id: assertion 'self != NULL' failed_

I get this when trying to connect with Todoist:

_(com.github.alainm23.planner:2): WARNING **: 16:19:09.281: TodoistOAuth.vala:125: Loading uri 'https://todoist.com/oauth/authorize?client_id=b0dd7d3714314b1dbbdab9ee03b6b432&scope=data:read_write,data:delete,project:delete&state=XE3K-4BBL-4XLG-UDS8' failed, error : Unacceptable TLS certificate_



I did some Googling and found this:

https://aur.archlinux.org/packages/appmenu-gtk-module-git/

Basically saying it's incompatible with Firefox. I tried installing Chromium and using that as my defaut but I get the same error

**Comment by alainm23** [comment]
@jleseane23  The error is `Unacceptable TLS certificate`, Are you using a VPN server?

**Comment by jleseane23** [comment]
Nope.

I even tried switching to different wifi networks to see if it was something on my end. Same result

**Comment by alainm23** [comment]
Enter this link in your browser anonymously, does it open normally?

https://todoist.com/oauth/authorize?client_id=b0dd7d3714314b1dbbdab9ee03b6b432&scope=data:read_write,data:delete,project:delete&state=XE3K-4BBL-4XLG-UDS8

**Comment by jleseane23** [comment]
It does. Both Chromium & Firefox

**Comment by alainm23** [comment]
@jleseane23 [this](https://valadoc.org/webkit2gtk-4.0/WebKit.WebContext.set_tls_errors_policy.html) could be a solution.
Can you help me to solve this problem please?

```
git clone https://github.com/alainm23/planner.git
cd planner/build-aux/flatpak
flatpak-builder --install repo com.github.alainm23.planner.json --force-clean --user
flatpak run com.github.alainm23.planner

```

**Comment by jleseane23** [comment]
I'm running the commands now. I had to install io.elementary.BaseApp & sdk org.gnome.sdk 3.36.

The compilation failed with the following:

../src/Services/Calendar/CalendarModel.vala:168.13-168.63: warning: copying delegates is not supported
Compilation succeeded - 14 warning(s)
[106/210] Compiling Vala source ../plugins/ca...aldav/Widgets/TaskRow.vala src/codecore.vapi.
FAILED: plugins/caldav/c41a7ba@@caldav@sha/caldav.c plugins/caldav/c41a7ba@@caldav@sha/Services/Store.c plugins/caldav/c41a7ba@@caldav@sha/Services/Utils.c plugins/caldav/c41a7ba@@caldav@sha/Widgets/SourceRow.c plugins/caldav/c41a7ba@@caldav@sha/Widgets/TaskList.c plugins/caldav/c41a7ba@@caldav@sha/Widgets/TaskRow.c plugins/caldav/caldav.h plugins/caldav/caldav.vapi 
valac -C --debug --debug --define=E_CAL_2_0 --pkg posix --pkg libical-glib --pkg libecal-2.0 --pkg libsoup-2.4 --pkg libedataserver-1.2 --pkg webkit2gtk-4.0 --pkg json-glib-1.0 --pkg sqlite3 --pkg libpeas-gtk-1.0 --pkg libpeas-1.0 --pkg granite --pkg gtk+-3.0 --pkg gee-0.8 --target-glib 2.30.0 --pkg glib-2.0 --color=always --directory plugins/caldav/c41a7ba@@caldav@sha --basedir ../plugins/caldav --library caldav --header plugins/caldav/caldav.h --vapi ../caldav.vapi ../plugins/caldav/caldav.vala ../plugins/caldav/Services/Store.vala ../plugins/caldav/Services/Utils.vala ../plugins/caldav/Widgets/SourceRow.vala ../plugins/caldav/Widgets/TaskList.vala ../plugins/caldav/Widgets/TaskRow.vala src/codecore.vapi
../plugins/caldav/Services/Store.vala:351.59-357.9: error: Cannot convert lambda expression to `GLib.Cancellable?'
../plugins/caldav/Services/Store.vala:351.53-351.56: warning: Argument 3: Cannot pass null to non-null parameter type
        client.modify_object.begin (comp, mod_type, null, (obj, res) => {
                                                    ^^^^
../plugins/caldav/Services/Store.vala:351.53-351.56: error: Argument 3: Cannot convert from `unowned null' to `uint32'
        client.modify_object.begin (comp, mod_type, null, (obj, res) => {
                                                    ^^^^
../plugins/caldav/Widgets/SourceRow.vala:213.9-219.9: warning: unhandled error `GLib.Error'
../plugins/caldav/Widgets/TaskRow.vala:477.13-477.95: error: Invalid assignment from owned expression to unowned variable
            remove_prop = ical_task.get_first_property (ICal.PropertyKind.DESCRIPTION_PROPERTY);



Compilation failed: 3 error(s), 2 warning(s)
[109/210] Compiling C object 'src/25a6634@@co...nerated_Services_Calendar_CalendarModel.c.o'.
ninja: build stopped: subcommand failed.
Error: module planner: Child process exited with code 1
`
I'm fairly new to Linux but whatever you need me to do let me know.

**Comment by alainm23** [comment]
@jleseane23 Clone the project again and build Planner again please.

**Comment by jleseane23** [comment]
Got to the Todoist auth window and was able to sign in with my Apple ID. Here's what I get now:

![image](https://user-images.githubusercontent.com/55364887/103184637-0402a880-4887-11eb-9d1e-4bd8ee1f0761.png)

Here's what's in Terminal:

* (com.github.alainm23.planner:2): WARNING **: 21:02:26.309: TodoistOAuth.vala:126: Loading uri 'https://github.com/alainm23/planner?code=93ff10ce3d29d3991ef5d29cf620d36f2440df03&state=XE3K-4BBL-4XLG-UDS8' failed, error : Load request cancelled

**Comment by alainm23** [comment]
@jleseane23 Clone the project again and build Planner again please.

**Comment by jleseane23** [comment]
- I got past the request page fail error in Planner.
- Terminal still shows but load request cancelled error, but Planner says it completed a sync.
- All of my labels didn't sync from Todoist
- Terminal shows the following:

(com.github.alainm23.planner:2): Json-CRITICAL **: 10:52:56.488: json_object_get_int_member: assertion 'JSON_NODE_TYPE (node) == JSON_NODE_VALUE' failed

(com.github.alainm23.planner:2): Json-CRITICAL **: 10:52:56.493: json_object_get_null_member: assertion 'node != NULL' failed

** (com.github.alainm23.planner:2): CRITICAL **: 10:52:57.262: objects_project_get_id: assertion 'self != NULL' failed

(com.github.alainm23.planner:2): Json-CRITICAL **: 11:07:58.834: json_object_get_object_member: assertion 'node != NULL' failed



![image](https://user-images.githubusercontent.com/55364887/103227534-8e392400-48fc-11eb-8f24-f053a19c3978.png)

![image](https://user-images.githubusercontent.com/55364887/103227555-9f823080-48fc-11eb-80c6-ee0e054d9566.png)

**Comment by alainm23** [comment]
Great, Sadly labels syncing is not available yet.

**Comment by jleseane23** [comment]
Ok cool. Thanks for all of your help and hard work. I really appreciate it.
